src/reuse/lint.py

```diff
@@ -252,12 +252,24 @@ def custom_serializer(obj):
             f"Object of type {obj.__class__.__name__} is not JSON serializable"
         )
 
+    def custom_sort_key(key):
+        order = {
+            "lint_version": 0,
+            "reuse_spec_version": 1,
+            "reuse_tool_version": 2,
+        }
+        return (order.get(key, 3), key)
+
+    report_dict = report.to_dict_lint()
+    sorted_report_dict = dict(
+        sorted(report_dict.items(), key=lambda item: custom_sort_key(item[0]))
+    )
```

Comment:
Instead of sorting it _here_, can we sort it inside of `to_dict_lint`?

And then write a test for `to_dict_lint` to see whether it's sorted?

Reply:
Done :)